code/game/machinery/iv_drip.dm

```diff
@@ -1,224 +1,684 @@
+			if(!L)
+				src.visible_message(SPAN_NOTICE("\The [src] buzzes, automatically deactivating \the [tank]."))
+				playsound(src, 'sound/machines/buzz-two.ogg', 50)
+				tank_off()
+				update_icon()
+				return
+			if(!tank_active) // Activates and sets the kPa to a safe pressure. This keeps from it constantly resetting itself
+				tank.distribute_pressure = safe_pressure_min
+				src.visible_message(SPAN_NOTICE("\The [src] chimes and adjusts \the [tank]'s release pressure"))
+				playsound(src, 'sound/machines/chime.ogg', 50)
+				tank_active = TRUE
+			if(L.checking_rupture == FALSE) // Safely retracts in case the lungs are about to rupture
+				src.visible_message(SPAN_WARNING("\The [src]'s flashes a warning light, automatically deactivating \the [tank] and retracting \the [breath_mask]."))
+				playsound(src, 'sound/machines/twobeep.ogg', 50)
+				breather.remove_from_mob(breath_mask)
+				breather.update_inv_wear_mask()
+				breath_mask.forceMove(src)
+				breath_mask.canremove = TRUE
+				tank_off()
+				update_icon()
+				return
+			if(tank.air_contents.return_pressure() == 0)
+				src.visible_message(SPAN_WARNING("\The [src] buzzes and automatically closes the valve."))
+				playsound(src, 'sound/machines/buzz-two.ogg', 50)
+				tank_off()
+				update_icon()
+				return
+			if(epp) // Emergency Positive Pressure system forces respiration
+				if(breather.losebreath > 0)
+					if(!epp_active)
+						src.visible_message(SPAN_WARNING("\The [src] flashes a blue light, activating it's Emergency Positive Pressure system!"))
+						playsound(breather, 'sound/machines/windowdoor.ogg', 50)
+						epp_active = TRUE
+						update_icon()
+					tank.distribute_pressure = safe_pressure_min // Constantly adjusts the pressure to keep up with the damage
+					breather.losebreath = 0
+					to_chat(breather, SPAN_NOTICE("You feel fresh air being pushed into your lungs."))
+			update_icon()
 
-	if (istype(W, /obj/item/reagent_containers/blood/ripped))
-		to_chat(user, "You can't use a ripped bloodpack.")
-		return
-	if (istype(W, /obj/item/reagent_containers))
-		if(!isnull(src.beaker))
-			to_chat(user, "There is already a reagent container loaded!")
+	if(istype(attached)) // IV attached
+		if(!beaker)
+			return
+		if(!attached.dna)
+			return
+		if(NOCLONE in attached.mutations)
+			return
+		if(attached.species.flags & NO_BLOOD)
 			return
 
-		user.drop_from_inventory(W,src)
-		src.beaker = W
-		to_chat(user, "You attach \the [W] to \the [src].")
-		src.update_icon()
-		return
-	else
-		return ..()
+		if(!attached.Adjacent(src))
+			iv_rip()
+			update_icon()
+			return
 
+		if(mode) // Injecting
+			if(beaker.reagents.total_volume > 0)
+				beaker.reagents.trans_to_mob(attached, transfer_amount, CHEM_BLOOD)
+				update_icon()
+			if(toggle_stop) // Automatically detaches if the blood volume is at 100%
+				if((beaker.reagents.has_reagent(/decl/reagent/blood) || beaker.reagents.has_reagent(/decl/reagent/saline)) && attached.get_blood_volume() >= 100)
```

Comment:
Not sure if it _needs_ to check the contents of the bag. It's an IV drip, not a reagent scanner. What if someone wants a coffee IV / something more nefarious? Or a medic wants a drug cocktail IV.

Reply:
The toggle stop is only for blood/saline and only if the person's blood volume is at 100%. Otherwise the IV will inject anything and stuff